sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -138,7 +142,9 @@ class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path:
   def applyFnToBatchByStream[RET](batchId: Long)(fn: InputStream => RET): RET = {
     val batchMetadataFile = batchIdToPath(batchId)
     if (fileManager.exists(batchMetadataFile)) {
-      val input = fileManager.open(batchMetadataFile)
```

Comment:
We encountered network blips on this open operation. This also seems safer to add retrying. I don't add retrying to other operations to avoid change current behaviors.